### PR TITLE
[Bug] Fix potenial bug when the index reading offset is greater than data length

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?

Fix potenial bug when the index reading offset equals to data length

### Why are the changes needed?

If the reading length == data length in `FixedSizeSegmentSplitter`, it should involve this block. And if its length > data length. it should abort it.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UTs

### CODE REVIEW

**Comment by zuston** [comment]
@leixm Could you help whether is a bug?  CC @jerqi

**Comment by leixm** [comment]
Here's an extra judgment. 
```
if (bufferOffset > 0) {
      ShuffleDataSegment sds = new ShuffleDataSegment(fileOffset, bufferOffset, bufferSegments);
      dataFileSegments.add(sds);
    }
```

**Comment by zuston** [comment]
You could use the test case to check this @leixm . It will fail.

**Comment by jerqi** [comment]
cc @kaijchen

**Comment by jerqi** [comment]
Do you find this problem in your online environment? Maybe we should also add some logs.

**Comment by leixm** [comment]
> You could use the test case to check this @leixm . It will fail.

You're right, it's a potenial bug.

**Comment by zuston** [comment]
> Do you find this problem in your online environment? 

No. I just found this bug when browsing recent commits. 

> Maybe we should also add some logs.

Good ideas. I will add some logs

File: common/src/test/java/org/apache/uniffle/common/segment/FixedSizeSegmentSplitterTest.java [hunk 1]
```diff
@@ -33,6 +33,28 @@
 
 public class FixedSizeSegmentSplitterTest {
 
+  @Test
+  public void testAvoidEOFException() {
```
Comment (kaijchen) [inline common/src/test/java/org/apache/uniffle/common/segment/FixedSizeSegmentSplitterTest.java:37]: This test could be changed to a ParameterizedTest.

**Review by kaijchen — COMMENTED** [review]
Thanks @zuston for the improvement, this change looks good.
Please remove the `[WIP]` tag in title if there is nothing to add.

File: common/src/test/java/org/apache/uniffle/common/segment/FixedSizeSegmentSplitterTest.java (hunk 1 above)
Comment (zuston) [reply common/src/test/java/org/apache/uniffle/common/segment/FixedSizeSegmentSplitterTest.java]: Done

**Review by jerqi — APPROVED** [review]
LGTM @leixm @kaijchen Do you have any other suggestion?

**Review by kaijchen — APPROVED** [review]
LGTM

**Comment by leixm** [comment]
LGTM.

**Comment by jerqi** [comment]
Merged. Thanks @zuston @kaijchen @leixm

**Pull request merged**